File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/FactoryMethod/Factory/WindowsDialog.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.FactoryMethod
{
    public class WindowsDialog : Dialog
    {
        public override IButton CreateButton()
        {
            return new WindowsButton();
        }
    }
}

File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/factorymethod/factory/WindowsDialog.java
package com.ute.refactoring.factorymethod.factory;

import com.ute.refactoring.factorymethod.buttons.Button;
import com.ute.refactoring.factorymethod.buttons.WindowsButton;

public class WindowsDialog extends Dialog{
    @Override
    public Button createButton() {
        return new WindowsButton();
    }
}
